cadc-util/src/main/java/ca/nrc/cadc/auth/AuthenticationUtil.java

```diff
@@ -350,6 +350,10 @@ public static Subject getSubject(PrincipalExtractor principalExtractor, boolean
 
         Subject subject = new Subject(false, principals, publicCred, privateCred);
         subject = validateSubject(subject);
+        // remove all un-validated authorization
+        for (AuthorizationTokenPrincipal tp : subject.getPrincipals(AuthorizationTokenPrincipal.class)) {
+            subject.getPrincipals().remove(tp);
+        }
```

Comment:
I don't understand the reason here.

Reply:
if the validate() step above did not process an AuthorizationTokenPrincipal (into an AuthorizationToken) it needs to be dropped because it is a raw input from user (via authorization header) and unknown to the server side code... it may break downstream code that looks at the set of principals because it will be an unrecognized type of principal.

just a missing safety measure needed because of change in ServletPrincipalExtractor.